drill.cpp

```diff
@@ -338,11 +338,23 @@ void ExcellonProcessor::export_ngc(const string of_dir, const boost::optional<st
 
                         if( nog81 )
                         {
+                            // Move to position
                             of << "G0 X"
                                << ( get_xvalue(x) - xoffsetTot ) * cfactor
-                               << " Y" << ( ( y - yoffsetTot ) * cfactor) << "\n";
-                            of << "G1 Z" << driller->zwork * cfactor << '\n';
-                                of << "G1 Z" << driller->zsafe * cfactor << '\n';
+                               << " Y" << ( ( y - yoffsetTot ) * cfactor) 
+                               << " F" << driller->g0_horizontal_speed * cfactor << "\n";
+                               
+                            // Plunge into the workpiece
+                            of << "G1 Z" << driller->zwork * cfactor 
+                               << " F" << driller->feed * cfactor << '\n';
+                            
+                            // Pull the drillbit out of the workpiece
+                            of << "G1 Z0"  
+                               << " F" << driller->feed * cfactor << '\n';
```

Comment:
As I understand, this code will cause the CNC to backout of the drilled hole first slowly and then quickly, right?

However, the G81 gcode always just backs out quickly.  So maybe you can delete these two lines?

https://www.cnccookbook.com/g81-g73-g83-drill-peck-canned-cycle/

Reply:
I believe you are correct, but I am not a machinist. My understanding is the more advanced machines have such commands. Everything I've worked with is all G0 / G1. I don't think this is inherently bad, but what it really comes down to is I have no "real" machines to test something like a G81 cycle with.